### ISSUE TITLE
[Bug] 'Go To Next Level' fax closes and disappears accidentally.

### ISSUE THREAD

**Issue by ARGAMX** [CorsixTH/CorsixTH#2638]
### Describe the issue

<img width="331" alt="Screenshot 2024-09-26 at 22 31 06" src="https://github.com/user-attachments/assets/4bc6b893-cae7-44e1-8936-bb703ff97574">

If you received 'Go To Next Level' fax and pressed any of these keys:
`1 2 3 4 5 Q Z C F T G B P M`
then fax closes and disappears.

pressing these keys:
`6 7 8 9 0 W E A S D X V Y H N U J I K O L SPACE ENTER `
do not causes this problem.

At the same time all other type of faxes do not disappears on any key press.
They can be closed on Esc press, but not disappears.

In original game 'Go To Next Level' letter do not closes on any key press. Even on Esc press.

### Steps to Reproduce

1. *Load save game attached to this ticket*
2. Press 5 or hold Z to change speed to max.
3. 1 Oct you will receive 'Go To Next Level' fax which opens automatically.
4. If you was holding Z fax will closes and disappears instantly.
5. If you was not holding Z, press 1 or P to slowdown the time to calmly read fax.
6. Observe, that it closes and disappears without any selection.

### Expected Behaviour

The "Go to the next level" fax should not close until the user selects any of the options. As in original game.

### Save Game

[finish_level7_save.sav.zip](https://github.com/user-attachments/files/17155282/finish_level7_save.sav.zip)

### CorsixTH Version

0.68.0 Beta 1

### Operating System

MacOS Sonoma 14.5

**Comment by lewri** [comment]
I'm impressed you took the dedication to testing every key 😅

**Comment by lewri** [comment]
Looks like the key handlers for the bottom_panel are interfering (`Q C F T G B M`). As well as the global speed ones (`1 2 3 4 5 P Z`.
`Esc` and `Q` are working as intended though:
```lua
  global_window_close = "escape",
  global_window_close_alt = "q",
```

Not sure why the others are causing issues.

**Comment by ARGAMX** [comment]
I have investigated a bit and what I discover:

1. Faxes of any type closes on press any of these keys:
`1 2 3 4 5 Q Z C F T G B P M`
2. After that accidentally close you can reopen any fax, except winGameFax.

Why player can't reopen winGameFax? 
Because it have timeout equal to 0, while other types of faxes have a timeout of several weeks.

Fax queue function description:
https://github.com/CorsixTH/CorsixTH/blob/d0fa5b78d1e2b2295261b096c27e44a585641254/CorsixTH/Lua/dialogs/bottom_panel.lua#L355-L355

https://github.com/CorsixTH/CorsixTH/blob/d0fa5b78d1e2b2295261b096c27e44a585641254/CorsixTH/Lua/dialogs/bottom_panel.lua#L336-L355

Win fax:
https://github.com/CorsixTH/CorsixTH/blob/939ddacd782535c10004e81e77f0f76e536d71b5/CorsixTH/Lua/world.lua#L1262

Some other type faxes:
https://github.com/CorsixTH/CorsixTH/blob/aaa1f5e61ea7a83c502920f4d1f4b92cd312b926/CorsixTH/Lua/hospitals/player_hospital.lua#L446

https://github.com/CorsixTH/CorsixTH/blob/aaa1f5e61ea7a83c502920f4d1f4b92cd312b926/CorsixTH/Lua/hospitals/player_hospital.lua#L724

But I can't understand what is the purpose of making 0 timeout for winGameFax?
User can can finish level only at certain date? No.

I think it made to be more similar to original Theme Hospital Win Letter.
But this Win Letter was non interruptible. User was unable to close or postpone it.

And since we don't have Win Letter yet, I don't see the reason to have that bug with disappearing Win Fax just to in order to be bit similar to original Theme Hospital Win Letter.

So this bug can be easily avoided just by setting 2-3 weeks timeout for this Win Fax.
I believe this compromise solution until we will develop Win Letter or make Fax be able to be non interruptible.

What does everyone think about this?
Is this an adequate solution?

**Comment by TheCycoONE** [comment]
Yes the user can only finish the level twice a year. It shouldn't let you unpause, that's a bug we have code to block it for other dialogs.

**Comment by ARGAMX** [comment]
Fax should be cancelled/closed only by the following actions:
1. Pressed `Esc`.
2. Pressed `Q`.
3. Pressed `M` (open/close message button).
4. Cursor press on icon in drawer.
5. Cursor press on 'X' button on fax screen.

(not actually these characters, but the hotkeys that default to those characters)
